.github/workflows/ci.yml

```diff
@@ -0,0 +1,64 @@
+name: Continuous integration
+on: [push, pull_request]
+
+jobs:
+  # This matrix job runs the test suite against multiple Ruby versions
+  test_matrix:
+    strategy:
+      fail-fast: false
+      matrix:
+        # Due to https://github.com/actions/runner/issues/849, we have to use quotes for '3.0'
+        ruby: [2.7, '3.0', 3.1]
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: ruby/setup-ruby@v1
+      with:
+        ruby-version: ${{ matrix.ruby }}
+        bundler-cache: true
+    - uses: actions/setup-node@v3.0.0
+      with:
+        node-version: lts/* # use the latest LTS release
+        cache: yarn
+    - run: yarn install
+    - run: bundle exec rake
+
+  # This job is needed to work around the fact that matrix jobs spawn multiple status checks – i.e. one job per variant.
+  # The branch protection rules depend on this as a composite job to ensure that all preceding test_matrix checks passed.
+  # Solution taken from: https://github.community/t/status-check-for-a-matrix-jobs/127354/3
+  test:
+    needs: test_matrix
+    runs-on: ubuntu-latest
+    steps:
+      - run: echo "All matrix tests have passed 🚀"
+
+  release:
+    needs: test
+    runs-on: ubuntu-latest
+    if: ${{ github.ref == 'refs/heads/master' }}
+    permissions:
+      contents: write
+    steps:
+    - uses: actions/checkout@v3
+    - uses: ruby/setup-ruby@v1
+      with:
+        rubygems: latest
+    - uses: actions/setup-node@v3.0.0
+      with:
+        node-version: lts/* # use the latest LTS release
+    - run: yarn install
+    - env:
+        GEM_HOST_API_KEY: ${{ secrets.ALPHAGOV_RUBYGEMS_API_KEY }}
+      run: |
+        VERSION=$(ruby -e "puts eval(File.read('govuk_admin_template.gemspec')).version")
+        GEM_VERSION=$(gem list --exact --remote govuk_admin_template)
+
+        if [ "${GEM_VERSION}" != "govuk_admin_template (${VERSION})" ]; then
+          gem build govuk_admin_template.gemspec
+          gem push "govuk_admin_template-${VERSION}.gem"
+        fi
+
+        if ! git ls-remote --tags --exit-code origin v${VERSION}; then
+          git tag v${VERSION}
+          git push --tags
+        fi
```

Comment:
Not a job for today, but I wonder if we can quickly set-up our action that does this for us so we don't have to repeat it everywhere.

Reply:
It looks like somebody has already done this. Although it doesn't seem to be very popular, so not sure if it's something we'd want to use.

https://github.com/marketplace/actions/gem-auto-tag

Reply:
Yeah - we're also quite sensitive about what workflows we use of others (https://gds-way.cloudapps.digital/standards/source-code.html#using-github-actions-and-workflows) due to the damage one could do with us mis-trusting one. So for some of the simpler cases it may be more pragmatic, if inefficient, to define our own.